pkg/apis/cache/v1alpha1/rediscluster_types.go

```diff
@@ -82,6 +85,11 @@ func (c *RedisCluster) ClassRef() *corev1.ObjectReference {
 	return c.Spec.ClassRef
 }
 
+// ConnectionSecretName return connection secret checking for overrides
+func (c *RedisCluster) ConnectionSecretName() string {
+	return util.IfEmptyString(c.Spec.ConnectionSecretNameOverride, c.Name)
```

Comment:
I understand this util function already exists and is used throughout the codebase, but I'd prefer we worked toward removing it. `util.IfEmptyString` tells me nothing about what this does. If empty string then what?

```go
func IfEmptyString(s, r string) string {
	if s == "" {
		return r
	}
	return s
}
```

This logic takes two seconds to write, is hard to get wrong, and documents its intention to the reader much better than `IfEmptyString`. I feel this is a clear case of "[a little copying is better than a little dependency](https://go-proverbs.github.io/)".